File: scenes/screensavers/wave/light.py
import pygame
import math
from .point3d import Point3D

class Light() :
	def __init__(self, x= -100, y = -100, z = -100, brightness = 1, window=None) :
		self.x = x
		self.y = y
		self.z = z
		self.brightness = brightness
		self.color = '#ffffff'
		self.point = Point3D(x,y,z)
		self.window = window

	def setPoint(self, x, y, z):
		self.x = x
		self.y = y
		self.z = z
		self.point.x = x
		self.point.y = y
		self.point.z = z

	def brightness(self, b) :
		self.brightness = math.max(b, 0)
		self.brightness = math.min(self.brightness, 1)
	
	def brightness(self) :
		return self.brightness

	def draw(self):
		x = self.point.screenX
		y = self.point.screenY
		self.render = pygame.draw.circle(self.window, self.color, [x, y], 10, 0)

File: web/client/src/js/lib/light.js
import Point3d from './point3d';

export default class Light {
	constructor(x, y, z, brightness) {
		this.x = (x === undefined) ? -100 : x;
		this.y = (y === undefined) ? -100 : y;
		this.z = (z === undefined) ? -100 : z;
		this.brightness = (brightness === undefined) ? 1 : brightness;
		this.color = "#ffffff";
		this.lineWidth = 1;
		this.alpha = 1;

		this.point = new Point3d(x, y, z);
	}

	setPoint(x,y,z) {

		this.x = x;
		this.y = y;
		this.z = z;

		this.point.x = x;
		this.point.y = y;
		this.point.z = z;
	}

	setBrightness(b) {
		this.brightness = Math.min(Math.max(b, 0), 1);
	}

	draw(context) {
		context.save();
		// context.lineWidth = this.lineWidth;
		context.fillStyle = this.color;
		context.beginPath();
		context.arc(this.point.getScreenX(), this.point.getScreenY(), 10, 0, 2 * Math.PI);
		// context.moveTo(this.pointA.getScreenX(), this.pointA.getScreenY());
		context.closePath();
		context.fill();
		// if (this.lineWidth > 0) {
		// 	context.stroke();
		// }
		context.restore();
	}
}
